test/integration/scheduler/scheduler_test.go

```diff
@@ -269,4 +273,70 @@ var _ = ginkgo.Describe("Scheduler", func() {
 			return createdJob.Spec.Suspend
 		}, framework.Timeout, framework.Interval).Should(gomega.Equal(pointer.Bool(false)))
 	})
+
+	ginkgo.It("Should schedule jobs according to the priority", func() {
+		// Initialize clusterQueue first but do not create
+		clusterQ := testing.MakeClusterQueue("cluster-queue").Resource(testing.MakeResource(corev1.ResourceCPU).
+			Flavor(testing.MakeFlavor(onDemandFlavor, "1").Obj()).Obj()).Obj()
+
+		queue := testing.MakeQueue("queue", ns.Name).ClusterQueue(clusterQ.Name).Obj()
+		gomega.Expect(k8sClient.Create(ctx, queue)).Should(gomega.Succeed())
+
+		highPriorityClass := testing.MakePriorityClass(highPriorityClassName).
+			PriorityValue(int32(highPriority)).Obj()
+		gomega.Expect(k8sClient.Create(ctx, highPriorityClass)).Should(gomega.Succeed())
+
+		lowPriorityClass := testing.MakePriorityClass(lowPriorityClassName).
+			PriorityValue(int32(lowPriority)).Obj()
+		gomega.Expect(k8sClient.Create(ctx, lowPriorityClass)).Should(gomega.Succeed())
+
+		job1 := testing.MakeJob("job1", ns.Name).Queue(queue.Name).Request(
+			corev1.ResourceCPU, "1").PriorityClass(lowPriorityClassName).Obj()
+		gomega.Expect(k8sClient.Create(ctx, job1)).Should(gomega.Succeed())
+		job2 := testing.MakeJob("job2", ns.Name).Queue(queue.Name).Request(
+			corev1.ResourceCPU, "1").PriorityClass(highPriorityClassName).Obj()
+		gomega.Expect(k8sClient.Create(ctx, job2)).Should(gomega.Succeed())
+
+		ginkgo.By("checking the workload1 is created with priority and priorityName")
+		createdWorkload1 := &kueue.QueuedWorkload{}
+		lookupKey1 := types.NamespacedName{Name: "job1", Namespace: ns.Name}
+		gomega.Eventually(func() bool {
+			err := k8sClient.Get(ctx, lookupKey1, createdWorkload1)
+			return err == nil
+		}, framework.Timeout, framework.Interval).Should(gomega.BeTrue())
+		gomega.Expect(createdWorkload1.Spec.PriorityClassName).Should(gomega.Equal(lowPriorityClassName))
+		gomega.Expect(*createdWorkload1.Spec.Priority).Should(gomega.Equal(int32(lowPriority)))
+
+		ginkgo.By("checking the workload2 is created with priority and priorityName")
+		createdWorkload2 := &kueue.QueuedWorkload{}
+		lookupKey2 := types.NamespacedName{Name: "job2", Namespace: ns.Name}
+		gomega.Eventually(func() bool {
+			err := k8sClient.Get(ctx, lookupKey2, createdWorkload2)
+			return err == nil
+		}, framework.Timeout, framework.Interval).Should(gomega.BeTrue())
+		gomega.Expect(createdWorkload2.Spec.PriorityClassName).Should(gomega.Equal(highPriorityClassName))
+		gomega.Expect(*createdWorkload2.Spec.Priority).Should(gomega.Equal(int32(highPriority)))
+
+		// Create clusterQueue
+		gomega.Expect(k8sClient.Create(ctx, clusterQ)).Should(gomega.Succeed())
+		defer func() {
+			gomega.Expect(framework.DeleteClusterQueue(ctx, k8sClient, clusterQ)).ToNot(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("checking the job with low priority suspends")
+		createdJob1 := &batchv1.Job{}
+		gomega.Eventually(func() *bool {
```

Comment:
we need to use consistently not eventually since we expect no change in the suspend status.